tests/e2e/globals.js

```diff
@@ -0,0 +1,98 @@
+doc = `
+API Management Postman Test Runner
+Usage:
+  test-runner.js <username> <password> <apigee_environment> <base_url> <apikey> <token_app_url>
+  test-runner.js -h | --help
+  -h --help  Show this text.
+`
+
+const fs = require('fs')
+const docopt = require('docopt').docopt
+const puppeteer = require('puppeteer')
+
+function nhsIdLogin(username, password, apigee_environment, base_url, apikey, login_url, writeGlobals, writeEnvVariables) {
+  (async () => {
+    console.log('Oauth journey on ' + login_url)
+
+    const browser = await puppeteer.launch({
+      executablePath: process.env.CHROME_BIN || null,
+      args: ['--no-sandbox', '--headless', '--disable-gpu']
+    });
+
+    const page = await browser.newPage()
+    await page.goto(login_url, { waitUntil: 'networkidle2' })
+    await page.click('#start')
+    await page.waitForSelector('#idToken1')
+    await page.type('#idToken1', username)
+    await page.type('#idToken2', password)
+    await page.click('#loginButton_0')
+    await page.waitForNavigation()
+
+    let credentialsJSON = await page.$eval('body > div > div > pre', e => e.innerText)
+    let credentials = credentialsJSON.replace(/'/g, '"')
+    let credentialsObject = JSON.parse(credentials)
+
+    await browser.close();
+
+    writeGlobals(credentialsObject.access_token, apikey)
+    writeEnvVariables(base_url, apigee_environment)
+  })()
+}
+
+function writeGlobals(token, apikey) {
```

Comment:
The newman library allows you to pass globals and environment as objects, which is what we do for PDS. I appreciate this might be necessary due to the docker container, but it seems a bit of a long way round. It may be worth customising the docker image used such that the templating dance is avoided -- it's an extra place for something to go wrong.